Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/Utilities/ValidationUtil.cs

```diff
@@ -0,0 +1,305 @@
+        /// <param name="useBlockList">If true, ignores elements not typically used for input</param>
+        /// <returns>List of DependencyProperties</returns>
+        public static List<DependencyProperty> GetDependencyProperties(this FrameworkElement element, bool useBlockList)
+        {
+            List<DependencyProperty> dependencyProperties = new List<DependencyProperty>();
+
+            bool isBlocklisted = useBlockList &&
+                (element is Panel || element is Button || element is Image || element is ScrollViewer || element is TextBlock ||
+                 element is Border || element is Shape || element is ContentPresenter);
+
+            if (!isBlocklisted)
+            {
+                Type type = element.GetType();
+                FieldInfo[] fields = type.GetFields(BindingFlags.Static | BindingFlags.Public | BindingFlags.FlattenHierarchy);
+                foreach (FieldInfo field in fields)
+                {
+                    if (field.FieldType == typeof(DependencyProperty))
+                    {
+                        dependencyProperties.Add((DependencyProperty)field.GetValue(null));
+                    }
+                }
+            }
+
+            return dependencyProperties;
+        }
+
+        /// <summary>
+        /// Determines if the specified exception is un-recoverable.
+        /// </summary>
+        /// <param name="exception">The exception.</param>
+        /// <returns>True if the process cannot be recovered from the exception.</returns>
+        public static bool IsCriticalException(Exception exception)
+        {
+#if WINDOWS_UWP
+            return exception is OutOfMemoryException;
+#else
+            return (exception is OutOfMemoryException) ||
+                (exception is StackOverflowException) ||
+                (exception is AccessViolationException) ||
+                (exception is ThreadAbortException);
+#endif
+        }
+
+        /// <summary>
+        /// Gets a list of active bindings on the specified FrameworkElement.  Bindings are gathered
+        /// according to the same conditions BindingGroup uses to find bindings of descendent elements
+        /// within the visual tree.
+        /// </summary>
+        /// <param name="element">Root FrameworkElement to search under</param>
+        /// <param name="inheritedDataContext">DomainContext of the element's parent</param>
+        /// <param name="dataItem">Target DomainContext</param>
+        /// <param name="twoWayOnly">If true, only returns TwoWay Bindings</param>
+        /// <param name="useBlockList">If true, ignores elements not typically used for input</param>
+        /// <returns>List of active bindings on the specified FrameworkElement.</returns>
+        private static List<BindingInfo> GetBindingInfoOfSingleElement(this FrameworkElement element, object inheritedDataContext, object dataItem, bool twoWayOnly, bool useBlockList)
```

Comment:
Same here?

Reply:
Same. We prefered to have a standalone DataGrid assembly.